aten/src/ATen/mps/MPSGeneratorImpl.mm

```diff
@@ -37,6 +37,10 @@ Generator createMPSGenerator(uint64_t seed_val) {
   engine_.reset_state(seed);
 }
 
+void MPSGeneratorImpl::set_offset(uint64_t offset) {
```

Comment:
I'm sure we have an offset here no?
cc @kulinseth

Reply:
I read the code, but it was not clear to me.

Reply:
[Here](https://github.com/pytorch/pytorch/pull/98965/files#diff-3a29e57c9df0293adc5940e0bc806ad0252d1c8f9303e97112eeb35603e4fb8bR59) we update the Philox state. The use is as follows: 
1. We start with the initialized state 
2. Pass it to our random op, which returns the updated Philox state
3. And then we write out the [state.](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/mps/operations/Distributions.mm#L106)

We have a state maintained [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/mps/MPSGeneratorImpl.h#L16). You can add offset parameter to update the state.